### PR TITLE
Add DynamoDB table detect and correct pipeline

### PR DESCRIPTION
**Pull request by khushboo9024**

### CODE REVIEW

File: dynamodb/dynamodb_table_stale_data.fp [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+      default_action     = param.default_action
+      enabled_actions    = param.enabled_actions
+    }
+  }
+}
+
+pipeline "correct_one_dynamodb_table_stale_data_exceeding_max_age" {
+  title         = "Correct one DynamoDB table stale data exceeding max age"
+  description   = "Runs corrective action on an DynamoDB table stale data exceeding max age."
+  // documentation = file("./dynamodb/docs/correct_one_dynamodb_table_stale_data_exceeding_max_age.md")
```
Comment (graza-io) [inline dynamodb/dynamodb_table_stale_data.fp:172]: Shouldn't be commented out

File: dynamodb/dynamodb_table_stale_data.fp [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    default     = var.dynamodb_table_stale_data_exceeding_max_age_enabled_actions
+  }
+
+  step "pipeline" "respond" {
+    pipeline = detect_correct.pipeline.correction_handler
+    args = {
+      notifier           = param.notifier
+      notification_level = param.notification_level
+      approvers          = param.approvers
+      detect_msg         = "Detected Dynamodb table stale data  ${param.title} exceeding maximum age."
```
Comment (graza-io) [inline dynamodb/dynamodb_table_stale_data.fp:231]: Should be `DynamoDB` as per other references?

File: dynamodb/dynamodb_table_stale_data.fp [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+          value        = "delete_table"
+          style        = local.style_alert
+          pipeline_ref = local.aws_pipeline_delete_dynamodb_table
+          pipeline_args = {
+            table_name  = param.name
+            region      = param.region
+            cred        = param.cred
+          }
+          success_msg = "Deleted Dynamodb table ${param.title}."
+          error_msg   = "Error deleting Dynamodb table ${param.title}."
```
Comment (graza-io) [inline dynamodb/dynamodb_table_stale_data.fp:259]: Should be DynamoDB as per other references?

File: thrifty.fpvars.example [hunk 4]
```diff
@@ -230,4 +230,11 @@ vpc_eips_if_unattached_enabled_actions  = ["skip", "release"]
...
 vpc_nat_gateways_if_unused_default_action   = "notify"
-vpc_nat_gateways_if_unused_enabled_actions  = ["skip", "delete"]
\ No newline at end of file
+vpc_nat_gateways_if_unused_enabled_actions  = ["skip", "delete"]
+
+# DynamoDB table stale data exceeding max age
+dynamodb_table_stale_data_max_days = 90
+dynamodb_table_stale_data_exceeding_max_age_trigger_enabled  = false
+dynamodb_table_stale_data_exceeding_max_age_trigger_schedule = "15m"
+dynamodb_table_stale_data_exceeding_max_age_default_action   = "delete_table"
```
Comment (graza-io) [inline thrifty.fpvars.example:239]: Value should be `"notify"`, allowing the user to override it but we shouldn't outright delete anything automatically out of the box.

**Review by graza-io — CHANGES_REQUESTED** [review]
See comments.

Additionally @misraved - I feel like the wording is wrong here: "stale data exceeding max age"... perhaps reads better as "with stale data", thoughts?

**Comment by misraved** [comment]
@graza-io agreed, I think `with_stale_data` is self-explanatory and we should go with it 👍.

**Review by graza-io — APPROVED** [review]

**Pull request merged**